File: madhacks/backend/task.js
class Task {
  
    constructor(name, isMine, isPublic, coinValue) {
      this.name = name;
      this.isMine = isMine;
      this.isComplete = false;
      this.isPublic = isPublic;
      this.coinValue = coinValue;
    }
  
    static getCoins() {
      return Task.coinValue;
    }
  
    setCoinValue() {
      if (!this.isMine) {
        Task.coinValue *= 2;
      }
    }
  
    completeTask() {
      this.isComplete = true;
      this.person.addCoins(Task.coinValue);
    }
  
    toString() {
      return this.name + ": " + coinValue;
    }
  }
  

File: madhacks/backend/Task.java


public class Task {
  private String name;
  private static int coinValue;
  private boolean isMine;
  private boolean isComplete;
  private Profile person;
  private boolean isPublic;

  public Task(String name, boolean isMine, boolean isPublic)
  {
    this.name = name;
    this.isMine = isMine;
    coinValue = 10;
    isComplete = false;
    this.isPublic = isPublic;
    
  }

  public static int getCoins()
  {
    return coinValue;
  }

  private void setCoinValue()
  {
    if (!isMine)
    {
      coinValue = coinValue * 2;
    }
  }
  
  public void completeTask()
  {
    isComplete = true;
    person.addCoins(coinValue);
  }

  public String toString()
  {
    return name + ": " + coinValue  + "coins";
  }

}